OpenChange/MAPIStoreMailMessage.m

```diff
@@ -316,15 +316,23 @@ - (void) _fetchHeaderData
           if (i == 0)
              {
                ASSIGN (headerMimeType, mimeType);
-               ASSIGN (headerCharset, charset);               
                parameters = partParameters;
              }
-          else
+
+          if ([mimeType isEqualToString: @"text/plain"])
             {
-              /* We can only put one header charset so we will use utf-8
-                 when we have different charsets in the parts */
-              if (headerCharset != charset)
-                ASSIGN (headerCharset, @"utf-8");
+              //  utf-8 is always encoded in text/plain
+              ASSIGN (headerCharset, @"utf-8");              
```

Comment:
Yes, text plain has always encoded to utf8. For example, in older versions, before any multipart fix we had:


```
- (int) getPidTagBody: (void **) data
             inMemCtx: (TALLOC_CTX *) memCtx
{
  NSString *stringValue;
  int rc = MAPISTORE_SUCCESS;

  if (!bodySetup)
    [self _fetchBodyData];

  if ([headerMimeType isEqualToString: @"text/plain"])
    {
      stringValue = [bodyContent bodyStringFromCharset: headerCharset];
      *data = [stringValue asUnicodeInMemCtx: memCtx];
    }
    (---)
```
And if you looked to  asUnicodeInMemCtx: you see the inside it was encoded to UTF8:
`      encoded = [self dataUsingEncoding: NSUTF8StringEncoding];`

So we have always encoded text-plain to utf-8, however in previous version the headerCharset could be set to the "original" text/plain charset. However we havent detected any problem in a no-multipart text/plain for it.

The problem in the previous attempt is that a mixed message could be assembled with parts with different charset and with a headerCharset (== PidTagInternetCodePage) which matched only some of the parts.

Maybe it would be worth to preserve text/plain charsets but since it is not an actual source of error I would let it slide to another time or PR.